audio/filter/af_lavcac3enc.c

```diff
@@ -283,10 +284,14 @@ static const struct mp_filter_info af_lavcac3enc_filter = {
 
 static void add_chmaps_to_autoconv(struct mp_filter *f,
                                    struct mp_autoconvert *conv,
-                                   const struct AVCodec *codec)
+                                   const AVCodecContext *avctx)
 {
-    const AVChannelLayout *lch = codec->ch_layouts;
-    for (int n = 0; lch && lch[n].nb_channels; n++) {
+    const AVChannelLayout *lch;
+    int ret = mp_avcodec_get_supported_config(avctx, NULL,
+                                              AV_CODEC_CONFIG_CHANNEL_LAYOUT,
+                                              (const void **)&lch);
```

Comment:
I would rather use this idiom:
```c
T *foo = NULL;
int ret = mp_avcodec_get_supported_config(...);
if (ret <= 0)
    foo = NULL;
```

alternatively we could integrate it into the wrapper so that we don't even need to check both the ret value and the output value

Reply:
> I would rather use this idiom:

It hides the intent of the API. NULL value means, that all formats are available, `ret < 0` means that there were some kind of error.

> alternatively we could integrate it into the wrapper so that we don't even need to check both the ret value and the output value

NULL is valid output from the API. I need to be checked on the user side.

Reply:
well that's annoying. the fact that this api can now fail I mean.